### ISSUE TITLE
using <b> tag will break the line

### ISSUE THREAD

**Issue by xareelee** [jsdf/react-native-htmlview#114]
Using `<b>` tag inside a string will break the line.

```js
<HTMLView
  value={ "A word will be <b>bold</b> inside this string." }
/>
```

* Expected:

> `A word will be `**`bold`**` inside this string."`

* What I got acctually with **react-native-htmlview**:

> `A word will be `
> **`bold`**
> ` inside this string."`

By the way, is there any convenient tag to add attributes to specific words in a string?

**Comment by agustind** [comment]
Had the same problem, fixed it like this:

`<HTMLView style={{flex:1, flexDirection: 'row', alignItems: 'flex-start', flexWrap: 'wrap'}} value={myhtmlcontent} />`

**Comment by AntAreS24** [comment]
Looking at the code, there is a prop called `addLineBreaks`. Try to set it to false?

**Comment by richchurcher** [comment]
Yes, that behaviour's probably not ideal right now. It's not just `<b>`: even a `<span>` will reproduce this. Note that it's not an issue if you can wrap the content in a block-level tag:
```
  <div>A word will be <b>bold</b> inside this string.</div>
```
I imagine there's a reasonable way to correct this, but for now it'll probably require a wrapper.

**Comment by isilher** [comment]
Alternatively, since `0.10` you can set `RootComponent={Text}` to make the content display inline style. Note that this can break when the html contains images. (can't render image within text component).

A side effect of both this and the wrapping `<div>` approach is that you'll lose line breaks between paragraphs etc.

**Comment by isilher** [comment]
Wrapping paragraphs in a `<p>` tag works too, if you have control over the HTML source.

**Comment by richchurcher** [comment]
I think there's probably a solution somewhere with wrapping any nodes that don't have a parent in a `Text` component to allow `flexDirection: 'row'`... which would also probably make handling block level elements easier, since they'd just be shunted to the next line in the flex column, no need for `\n`. Just not quite sure how to do that with htmlparser2 yet... :thinking:

**Comment by laurasantana** [comment]
This issue keeps happening and none of the solutions above fix it for me

**Comment by jaschaio** [comment]
@isilher I used your solution adding `RootComponent={Text}`, but I don't understand how to style the Text Component?

**Comment by dancherb** [comment]
Update: best way seems to be to pas in a custom component with RootComponent={MyText} that has the styling applied by default.

**Comment by necixy** [comment]
Although RootComponent={Text} worked for me but I could not live with seeing the warning. 😄 
Hence I created a custom component named HTMLText and wrapped the HTMLView inside this (because for my project I only needed to show the text most of the time), the value is also wrapped inside the <div>. I also created the 'style' prop so I could pass the style like fontSize from the component I'm going to reuse it in.

I admit, I could do this without making custom component but it easier to reuse multiple time. 

Below is the component code and the method of using it, hopefully it helps someone someday.

`import React, { Component } from "react";
import PropTypes from "prop-types";
import HTMLView from "react-native-htmlview";

class HTMLText extends Component {
  render() {
    return (
      <HTMLView
        value={`<div>${this.props.value}</div>`}
        stylesheet={{ div: this.props.style }}
      />
    );
  }
}

HTMLText.propTypes = {
  value: PropTypes.string.isRequired,
  style: PropTypes.number
};

export default HTMLText;
`


`<HTMLText
                        value="cor<b>r</b>ect-<i>title</i>-2"
                        style={{
                          fontSize: 30
                        }}
                      />`

**Comment by DeVoresyah** [comment]
hello,
I'm trying to wrap with `<div></div>`

but my image be like this:
![bug](https://i.imgur.com/b2WEARJ.png)

my code:
```renderNode(node, index, siblings, parent, defaultRenderer) {
        if (node.name == 'img') {
            const { src, height } = node.attribs;
            const imageHeight = height || 300;
            return (
                <Image
                key={index}
                style={{ width: width * PixelRatio.get(), height: imageHeight * PixelRatio.get(), overflow:'hidden', alignSelf:'center' }}
                source={{ uri: src }}
                resizeMode='contain'/>
            );
        }
    }

<HTMLView
                        value={`<div>${data.content}</div>`}
                        renderNode={this.renderNode}
                        stylesheet={htmlStyles}
                        style={{flexWrap:'nowrap', flexDirection:'column'}}
                        textComponentProps={{
                            style: {
                                fontFamily: Fonts.OpenSansLight,
                                fontSize: 15,
                                lineHeight: 25,
                                color: '#4E4E4E'
                            }
                        }}
                        onLinkPress={(url) => navigation.navigate('UrlView', {url})}
                    />

const htmlStyles = StyleSheet.create({
    p: {
        fontFamily: Fonts.OpenSansLight,
        color: '#4e4e4e',
        fontSize: 14,
        lineHeight: 20
    },
    h2: {
        fontFamily: Fonts.OpenSansSemiBold,
        fontSize: 18
    },
    div: {
        flexWrap: 'nowrap'
    }
})

**Comment by lichbt-rks** [comment]
I'm also encountering the same error. I don't know if there's a solution yet. I tried using the prop rootComponent={Text}, but when the user goes to the next line, it deletes the newline character. What's the solution?
